library/cartridge_instance.py

```diff
@@ -14,21 +15,115 @@
 }
 
 
+def serialize_dict(d):
+    parts = []
+
+    for k, v in d.items():
+        value_str = None
+        if type(v) == str:
+            value_str = "'%s'" % v
+        elif type(v) == float or type(v) == int:
+            value_str = "%s" % v
+        elif type(v) == bool:
+            value_str = 'true' if v else 'false'
+        else:
+            raise CartridgeException(
+                cartridge_errcodes.BAD_VALUE_TYPE,
+                "Unknown value type: {}" % type(v)
+            )
+
+        parts.append("%s = %s" % (k, value_str))
+
+    return '{ %s }' % ', '.join(parts)
+
+
+def box_cfg_was_called(control_console):
+    return control_console.eval('''
+        return type(box.cfg) ~= 'function'
+    ''')
+
+
+def get_box_cfg(control_console):
+    return control_console.eval('''
+        return box.cfg or box.NULL
+    ''')
+
+
+def change_memory_size(param_name, cartridge_defaults, config, control_console):
+    new_memory_size = None
+    if param_name in config:
+        new_memory_size = config[param_name]
+    else:
+        new_memory_size = cartridge_defaults[param_name]
+
+    if new_memory_size is None:
+        return False
+
+    # Get current memory size
+    current_memory_size = control_console.eval('''
+        return type(box.cfg) ~= 'function' and box.cfg.{} or box.NULL
```

Comment:
Are you sure `box.cfg{}` implicitly should be called here instead of return an error?

Reply:
So, it seems a typo here `box.cfg.{}` -> `box.cfg{}`